controllers/openstackcluster_controller.go

```diff
@@ -450,9 +453,27 @@ func (r *OpenStackClusterReconciler) SetupWithManager(ctx context.Context, mgr c
 		).
 		Watches(
 			&source.Kind{Type: &clusterv1.Cluster{}},
-			handler.EnqueueRequestsFromMapFunc(util.ClusterToInfrastructureMapFunc(infrav1.GroupVersion.WithKind("OpenStackCluster"))),
+			handler.EnqueueRequestsFromMapFunc(func(o client.Object) []reconcile.Request {
+				requests := clusterToInfraFn(o)
+				if requests == nil {
```

Comment:
```suggestion
				if len(requests) > 0 {
```
Should catch both cases and then we're safe against nil pointers in l.463

Reply:
https://github.com/kubernetes-sigs/cluster-api/blob/master/util/util.go#L195-L222
It either returns nil or an array of len == 1.. I could change it to 
```
if len(requests) < 1 {
   return nil
}
```

Reply:
Ups of course, accidentally inverted the logic :)